lib/charms/operator_libs_linux/v1/snap.py

```diff
@@ -976,15 +976,21 @@ def _system_set(config_item: str, value: str) -> None:
         raise SnapError("Failed setting system config '{}' to '{}'".format(config_item, value))
 
 
-def hold_refresh(days: int = 90) -> bool:
+def hold_refresh(days: int = 90, forever: bool = False) -> bool:
     """Set the system-wide snap refresh hold.
 
     Args:
         days: number of days to hold system refreshes for. Maximum 90. Set to zero to remove hold.
+        forever: if set  to True, will set a hold forever.
     """
+    if not isinstance(forever, bool):
+        raise ValueError("forever must be a boolean")
     # Currently the snap daemon can only hold for a maximum of 90 days
-    if not isinstance(days, int) or days > 90:
+    elif not isinstance(days, int) or days > 90:
```

Comment:
No need for `elif` as the above will have raised (I find it more confusing to keep track of when there's a run of "else"s.) Also, I think we should adjust the type check to raise `TypeError`, and the range check to also check for <0. My suggestion is that we simplify by doing the type checking up-front, and then the range checking where it's relevant:

```python
    if not isinstance(days, int):
        raise TypeError("days must be an int")
    if forever:
        ...
    elif days == 0:
        ...
    else:
        if not 1 <= days <= 90:
            raise ValueError("days must be between 1 and 90")
        ...
```